core/dbt/contracts/project.py

```diff
@@ -101,13 +101,26 @@ class PackageConfig(dbtClassMixin):
     @classmethod
     def validate(cls, data):
         for package in data.get("packages", data):
+            # This can happen when the target is a variable that is not filled and results in hangs
+            if isinstance(package, dict):
+                if package.get("package") == "":
+                    raise ValidationError(
+                        "A package is missing the value. It is a required property."
```

Comment:
Should we add some more detailed message here to help folks narrow down the problem since we already know what type is missing value here?